debian/patches/pop_upgrade.patch

```diff
@@ -292,3 +65,287 @@ Index: gnome-control-center/shell/cc-window.h
++    gtk_container_set_border_width (GTK_CONTAINER (upgrade_widget), 12);
++    gtk_container_add (GTK_CONTAINER (self->overlay), upgrade_widget);
++    pop_upgrade_widget_callback_error (POP_UPGRADE, (PopUpgradeWidgetErrorCallback) pop_upgrade_callback_error, self);
++}
++
++static void cc_upgrade_panel_dispose (GObject *object) {
++    CcUpgradePanel *prefs = CC_UPGRADE_PANEL (object);
++
++    // Detach the pop-upgrade widget on disposing this panel.
++    GtkWidget *pop = pop_upgrade_widget_container (POP_UPGRADE);
++    gtk_container_remove (GTK_CONTAINER (prefs->overlay), pop);
++
++    G_OBJECT_CLASS (cc_upgrade_panel_parent_class)->dispose (object);
++}
++
++static void cc_upgrade_panel_finalize (GObject *object) {
++    G_OBJECT_CLASS (cc_upgrade_panel_parent_class)->finalize (object);
++}
++
++static void cc_upgrade_panel_class_init (CcUpgradePanelClass *klass) {
++    GObjectClass *object_class = G_OBJECT_CLASS (klass);
++
++    object_class->dispose = cc_upgrade_panel_dispose;
++    object_class->finalize = cc_upgrade_panel_finalize;
++}
++
++static void cc_upgrade_panel_init (CcUpgradePanel *prefs) {
++    prefs->overlay = gtk_overlay_new ();
++    pop_os_info_overlay (prefs);
++    pop_os_upgrade_widget (prefs);
++
++    gtk_container_add (GTK_CONTAINER (prefs), prefs->overlay);
++}
+Index: gnome-control-center/panels/upgrade/cc-upgrade-panel.h
+===================================================================
+--- /dev/null
++++ gnome-control-center/panels/upgrade/cc-upgrade-panel.h
+@@ -0,0 +1,32 @@
++/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
++ *
++ * Copyright (C) 2019 System76
++ *
++ * This program is free software; you can redistribute it and/or modify
++ * it under the terms of the GNU General Public License as published by
++ * the Free Software Foundation; either version 2 of the License, or
++ * (at your option) any later version.
++ *
++ * This program is distributed in the hope that it will be useful,
++ * but WITHOUT ANY WARRANTY; without even the implied warranty of
++ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++ * GNU General Public License for more details.
++ *
++ * You should have received a copy of the GNU General Public License
++ * along with this program; if not, see <http://www.gnu.org/licenses/>.
++ *
++ */
++
++#pragma once
++
++#include <shell/cc-panel.h>
++
++extern gint POP_UPGRADE_STATUS;
++
++G_BEGIN_DECLS
++
++#define CC_TYPE_UPGRADE_PANEL (cc_upgrade_panel_get_type())
++G_DECLARE_FINAL_TYPE(CcUpgradePanel, cc_upgrade_panel, CC, UPGRADE_PANEL,
++                     CcPanel)
++
++G_END_DECLS
+Index: gnome-control-center/panels/upgrade/gnome-upgrade-panel.desktop.in.in
+===================================================================
+--- /dev/null
++++ gnome-control-center/panels/upgrade/gnome-upgrade-panel.desktop.in.in
+@@ -0,0 +1,14 @@
++[Desktop Entry]
++Name=OS Upgrade
++Comment=OS Upgrade
```

Comment:
Comment should read: `View system upgrade, recovery and refresh information` per discussion today with Maria.

Reply:
and done